src/python/RestClient/AuthHandling/X509Auth.py

```diff
@@ -82,16 +86,30 @@ def __search_cert_key(self):
         else:
             raise ClientAuthException("No valid X509 cert-key-pair found.")
 
+    def __create_ca_bundle(self):
+        trust_ca_files = glob.glob(f"{self._ca_path}/*.pem")
+        trust_ca_files.append(self._ssl_cert)
+        ca_bundle = f"/tmp/trust_ca_{os.getuid()}.crt"
+
+        with open(ca_bundle,"wb") as bundles:

```

Comment:
this code leaks user information to the system and to the publicly available directory. As such the `/tmp/trust_ca_{os.getuid()}.crt` will be created and live in /tmp area forever until system will be rebooted. Therefore, any intruder can steal user information. Instead, you should use temporary file and delete it once you create a bundle such that code will not leak user ssl cert to the underlying system.

Reply:
I resolved this request to use ```tempfile.NamedTemporaryFile```. X509Auth class has a class static variable to keep a *CA bundle temp file* until the program is terminated.